HealthMate/HealthMate.BLL/Services/MoodService.cs

```diff
@@ -0,0 +1,29 @@
+﻿using AutoMapper;
+using HealthMate.BLL.Abstractions;
+using HealthMate.BLL.Exceptions;
+using HealthMate.DAL.Abstractions;
+using HealthMate.DAL.Entities;
+
+namespace HealthMate.BLL.Services
+{
+    internal class MoodService(IMoodRepository moodRepository, IMapper mapper)
+        : GenericService<Mood, Models.Mood>(moodRepository, mapper), IMoodService
+    {
+        public async Task<Models.Mood> GetMoodByDate(DateTime date, CancellationToken token)
+        {
+            var moodEntity = await moodRepository.GetMoodByDate(date, token) ??
+                             throw new NotFoundException($"Mood on date: {date} not found.");
```

Comment:
Please, just return null for this.

Imagine situation when you're trying to switch between dates and backend every time return 404 instead of just null.